### PR TITLE
Subsystem refactor

### LINKED ISSUE #6: Rewrite our code to utilize an FRC-like subsystem structure
DONE:
- Created separate ROS nodes for each subsystem
- Figure out how to communicate between rovr_control and the subsystems (services!)
- Rewrote rovr_control to use services
- Rewrote motor_control to utilize services
- Tested this on my desktop in Ubuntu
- Updated the Lucid Chart diagram

IN-PROGRESS:
- Test this on last year's robot!

### PR DESCRIPTION
**Pull request by Isopod00**
Ready to merge! :)

### CODE REVIEW

File: src/conveyor/conveyor/conveyor_node.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+# Import the ROS 2 Python module
+import rclpy
+from rclpy.node import Node
+
+# Import custom ROS 2 interfaces
+from rovr_interfaces.srv import MotorCommandSet
+from rovr_interfaces.srv import ConveyorSetPower, Stop
+
+# Define helper functions here
+def clamp(number: float, minimum: float, maximum: float) -> float:
```
Comment (jmblixt3) [inline src/conveyor/conveyor/conveyor_node.py:15]: Find a way to not redfine this many times

Comment (jmblixt3) [reply src/conveyor/conveyor/conveyor_node.py:15]: Can make this into a separate ticket

File: src/conveyor/conveyor/conveyor_node.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        # Current subsystem state
+        self.running = False
+
+    # Define subsystem methods here
+    def set_power(self, drum_belt_power: float, conveyor_belt_power: float) -> None:
+        """This method sets power to the conveyor belts."""
+        self.running = True
+        drum_belt_power = clamp(drum_belt_power, -1.0, 1.0)  # Clamp the power between -1.0 and 1.0
+        conveyor_power = clamp(conveyor_belt_power, -1.0, 1.0)  # Clamp the power between -1.0 and 1.0
+        self.cli_motor_set.call_async(MotorCommandSet.Request(type="duty_cycle", can_id=self.DRUM_BELT_MOTOR, value=-1*drum_belt_power))
```
Comment (jmblixt3) [inline src/conveyor/conveyor/conveyor_node.py:46]: Ideally don't just multiply by -1, you can make an inverse constant, but ideally this is reversed in vesc config

Comment (jmblixt3) [reply src/conveyor/conveyor/conveyor_node.py:46]: This can be a new ticket

File: src/conveyor/package.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  <exec_depend>rclpy</exec_depend>
+  <exec_depend>std_msgs</exec_depend>
+  <exec_depend>ros2launch</exec_depend>
+  <exec_depend>rovr_interfaces</exec_depend>
+
+  <export>
+    <build_type>ament_python</build_type>
+  </export>
+</package>
```
Comment (jmblixt3) [inline src/conveyor/package.xml:27]: Normalize all file endings ticket needed

File: src/digger/digger/digger_node.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        if self.running:
+            self.stop()
+        else:
+            self.set_power(power)
+            
+    # Define service callback methods here
+    def set_power_callback(self, request, response) -> None:
+        """This service request sets power to the digging drum."""
+        self.set_power(request.power)
+        response.success = 0 # indicates success
```
Comment (jmblixt3) [inline src/digger/digger/digger_node.py:62]: Consider making this a constant/enum

File: src/drivetrain/drivetrain/drivetrain_node.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        self.cli_motor_set.call_async(MotorCommandSet.Request(type="duty_cycle", can_id=self.FRONT_RIGHT_DRIVE, value=right_power * -1))
+        self.cli_motor_set.call_async(MotorCommandSet.Request(type="duty_cycle", can_id=self.BACK_RIGHT_DRIVE, value=right_power * -1))
+
+    def stop(self):
+        """This method stops the drivetrain."""
+        self.drive(0.0, 0.0)
+        
+    # Define service callback methods here
+    def stop_callback(self, request, response) -> None:
+        """This service request stops the offloading belt."""
```
Comment (jmblixt3) [inline src/drivetrain/drivetrain/drivetrain_node.py]: Fix this comment

File: src/rovr_interfaces/package.xml
Comment (jmblixt3) [inline src/rovr_interfaces/package.xml:1]: Do these need to be separate from each node, if so I'm just dumb and tell me to close this

**Review by jmblixt3 — CHANGES_REQUESTED** [review]
Most of these are future tickets, but we need to fix a few simple things

File: src/drivetrain/drivetrain/drivetrain_node.py (hunk 5 above)
Comment (Isopod00) [reply src/drivetrain/drivetrain/drivetrain_node.py]: done

**Review by jmblixt3 — COMMENTED** [review]
All good to go

**Review by jmblixt3 — APPROVED** [review]

**Pull request merged**